diff --git a/train.py b/train.py
--- a/train.py
+++ b/train.py
@@ -74,4 +74,7 @@
         ("eval/episode_contact_force", "contact_force"),
+        ("eval/episode_joint_limit", "joint_limit"),
+        ("eval/episode_illegal_contact", "illegal_ct"),
         ("eval/episode_done_low_height", "done_low"),
         ("eval/episode_done_tipped", "done_tip"),
+        ("eval/episode_done_illegal_contact", "done_illegal"),
         ("eval/episode_done_invalid", "done_nan"),
