diff --git a/biomechanics_env.py b/biomechanics_env.py
--- a/biomechanics_env.py
+++ b/biomechanics_env.py
@@ -1314,2 +1314,6 @@
         contact_force_cost = self._get_contact_force_cost(data)
+        joint_limit = self._get_soft_joint_limit_violation(data)
+        joint_limit_cost = self.SOFT_JOINT_LIMIT_COST_SCALE * joint_limit
+        illegal_contact = self._get_illegal_contact(data)
+        illegal_contact_cost = self.ILLEGAL_CONTACT_COST_SCALE * illegal_contact
         foot_slip = self._get_foot_slip_cost(
@@ -1358,2 +1362,4 @@
             "contact_force": contact_force_cost,
+            "joint_limit": joint_limit_cost,
+            "illegal_contact": illegal_contact_cost,
             "foot_slip": foot_slip_cost,
@@ -1398,3 +1404,4 @@
         )
-        done = too_low | tipped_over | invalid
+        done_illegal_contact = self._get_fatal_illegal_contact(data)
+        done = too_low | tipped_over | invalid | done_illegal_contact
         reward = jp.where(done, self.FALL_REWARD, reward)
@@ -1431,4 +1438,7 @@
             "contact_force": contact_force_cost,
+            "joint_limit": joint_limit,
+            "illegal_contact": illegal_contact,
             "done_low_height": too_low,
             "done_tipped": tipped_over,
+            "done_illegal_contact": done_illegal_contact,
             "done_invalid": invalid,
