diff --git a/train.py b/train.py
--- a/train.py
+++ b/train.py
@@ -784,2 +784,6 @@
             "reference_target_observation": env_config.reference_target_observation,
+            "reference_phase_randomization": (
+                env_config.reference_phase_randomization
+            ),
+            "reference_state_init": env_config.reference_state_init,
             "action_smoothing": env_config.action_smoothing,
